tests/DEN_constructorTests.cpp

```diff
@@ -0,0 +1,59 @@
+#include <SpMV.hpp>
+#include <vector>
+
+// Testing library required for testing (Always include this last!)
+#include "unit_test_framework.hpp"
+
+TEST_CASE(test_DEN_constructor_double){
+    // test creation of new object
+    SpMV::SparseMatrix_DEN<double> A(5,6);
+
+    // check that the values assigned into the object are correct
+    ASSERT(A.nrows == 5);
+    ASSERT(A.ncols == 6);
+
+    // check that object state is initialized
+    ASSERT(A._state == SpMV::MatrixState::initialized)
+}
+
+TEST_CASE(test_DEN_destructor_double){
+    // test that object can be deleted
+    SpMV::SparseMatrix_DEN<double>* testing = new SpMV::SparseMatrix_DEN<double>(5,5); 
+    delete testing;
+    // Test will fail if the object can not be deleted
+    // If test fails, a segmentation fault will occur
+}
+
+TEST_CASE(test_DEN_constructor_int){
+    // test creation of new object
+    SpMV::SparseMatrix_DEN<int> A(5,6);
```

Comment:
We do not require to have support for `int` data types. Is this test case necessary?